openrouteservice/src/main/java/org/heigit/ors/routing/graphhopper/extensions/core/PrepareCore.java

```diff
@@ -509,11 +509,11 @@ public RoutingAlgorithm createAlgo(Graph graph, AlgorithmOptions opts) {
         String algoStr = ASTAR_BI;
 
         if (ASTAR_BI.equals(algoStr)) {
-            CoreALT tmpAlgo = new CoreALT(graph, prepareWeighting, traversalMode);
+            CoreALT tmpAlgo = new CoreALT(graph,new PreparationWeighting(opts.getWeighting()), traversalMode);
             tmpAlgo.setApproximation(RoutingAlgorithmFactorySimple.getApproximation(ASTAR_BI, opts, graph.getNodeAccess()));
             algo = tmpAlgo;
         } else if (DIJKSTRA_BI.equals(algoStr)) {
-            algo = new CoreDijkstra(graph, prepareWeighting, traversalMode);
+            algo = new CoreDijkstra(graph, new PreparationWeighting(opts.getWeighting()), traversalMode);
```

Comment:
The prepareWeighting variable is set in the constructor during preprocessing and it is final so it cannot be changed when using CALT. Since MaximumSpeedWeighting may or may not be used I need to allow for a dynamic change of the weighting using the code I wrote.